pa4/checker-pa4.ml

```diff
@@ -1212,121 +1241,101 @@ let main () = begin
         vertices := add_missing_edges !vertices;
         let sorted_classes = topo_sort !vertices in
 
-        (*List.iter (fun cls -> *)
-                (*printf "%s\n" cls*)
-        (*) sorted_classes ;*)
-
-        let does_method_exist_in_list target_method_name feature_list =
-                let result = ref false in
-                List.iter (fun feature ->
-                        match feature with
-                        | Attribute _ -> ()
-                        | Method ((_,method_name), _, _, _) -> begin
-                                if method_name = target_method_name then
-                                        result := true;
-                                        ()
-                        end
-                ) feature_list ;
-                !result
-        in
-
         let output_formal_name formals =
                 List.iter( fun ((_, formal_name),_) ->
                         fprintf f_out "%s\n" formal_name
                 ) formals                 
         in
 
+        let methods_map = ref (Hashtbl.create 255) in
+        let ordered_method_names = ref [] in
         (* retreive the hashtable with method name as a key and tuple (formals, class_name, exp) as a value.
          * It is needed to conveniently print out implementation map. 
          * What it does? it itrates over class names (that are in topological order) and saves all
          * methods that it declares and override.*)
         let get_parents_methods_tbl class_names super_class_list =
-            let methods_map = Hashtbl.create 255 in
-            List.iter (fun class_name ->
-                    let super_class_option = List.find_opt (fun ((_,cname),_,_)-> class_name = cname) super_class_list in
-                    if not (Option.none = super_class_option) then begin
-                            let ((_, _),_,features) = Option.get super_class_option in
-                            (* add methods to hashtbl starting from the ultimate class that override method *)
-                            List.iter (fun current_method ->
-                                    match current_method with
-                                    | Attribute _ -> ()
-                                    | Method ((_,method_name), formals, method_type, exp) -> begin
-                                            let cur = Hashtbl.find_opt methods_map method_name in
-                                            match cur with 
-                                            | None -> Hashtbl.replace methods_map method_name (formals, class_name, exp);
-                                            | _ -> () 
-                                    end
-                            ) features ;
-                    end;
-            ) class_names ;
-            methods_map
+                (*let methods_map_ = !methods_map in*)
```

Comment:
Same point about indention here. This should just be one level of indention rather than two.